teps/0037-simplify-metrics.md

```diff
@@ -0,0 +1,230 @@
+tekton_taskrun_duration_seconds_bucket{namespace="arendelle-nsfqw",status="success",task="anonymous",taskrun="duplicate-pod-task-run-wnigeayt",le="+Inf"} 1
+tekton_taskrun_duration_seconds_sum{namespace="arendelle-nsfqw",status="success",task="anonymous",taskrun="duplicate-pod-task-run-wnigeayt"} 122
+tekton_taskrun_duration_seconds_count{namespace="arendelle-nsfqw",status="success",task="anonymous",taskrun="duplicate-pod-task-run-wnigeayt"} 1
+```
+
+After
+
+```log
+tekton_taskrun_duration_seconds{namespace="arendelle-nsfqw",status="success",task="anonymous",taskrun="duplicate-pod-task-run-wnigeayt"} 122
+```
+
+we can add a `config-observability` option to switch `duration_seconds` type. The default value is `histogram`.
+It can be set to `gauge` if users only care about the execution time of individual `TaskRun`.
+It can't be set to `gauge` when `metrics.namespace-level` is `true` because gauge can't be aggregated at namespace level.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-observability
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+data:
+  metrics.duration-seconds-type: gauge
+```
+
+### Notes/Caveats
+
+The ultimate goal of this TEP is to provide the metrics that users truly need. So the `config-observability` option is not planned to exist indefinitely.
+Add a `config-observability` option to only report at the namespace level, default it to false for 1+ release, mention this in release
+notes. If there's no significant user pushback, default it to true for 1+ release, then remove it entirely.
+If there is users' feedback, it can be used to guide future decisions.
+
+## Alternatives
+
+### Alternatives 1: Customize metrics
+
+End users can also be allowed to customize what they want to collect and what level of granularity. We can add a flag to specify whether users want to customize the metrics.
+If it is set to `true`, Tekton pipelines would only report the metrics users configured.
+
+The configuration could be like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-observability
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+data:
+  metrics.customize-metrics: "true"
+  metrics.customize-metrics-spec: |
+    {
+      "metrics": [{
+          "name": "pipelinerun_duration_seconds",
+          "labels": ["status", "namespace"]
+        }, {
+          "name": "taskrun_duration_seconds",
+          "labels": ["status", "namespace"]
+        }, {
+          "name": "taskrun_count"
+        }
+      ]
+    }
+```
+
+If user doesn't specify the labels, use the default labels for the metrics. Besides, specifying every metrics users want might be annoying. We can only allow user to configure metrics whose type is histogram.
```

Comment:
as far as i can tell, a benefit of this proposal is we could still add this later if needed